### DISCUSSION TITLE
How is storage handled for containers and services?

### DISCUSSION THREAD

**Discussion by madalinignisca** [SquirrelCorporation/SquirrelServersManager#293] [Q&A] [answered]
Scenario: setup is done on a bunch of servers, example 7 servers.

As the project uses docker to deploy whatever the admin wants, how is storage handled?

Is it using only local storage (locking containers to the server they been started), or does it helps easy and fast to setup some replicated remote storage, example using Ceph or GlusterFS?

**Comment by SquirrelDeveloper** [comment, marked as answer]
Hi @madalinignisca . Do you mean the "Deploy" feature in "Containers" ? If so, only simple volumes are supported now, like in those screenshots
<img width="789" alt="Screenshot 2024-09-18 at 08 50 40" src="https://github.com/user-attachments/assets/bb598c89-14b3-470d-b0f2-da5728b59a1c">
<img width="725" alt="Screenshot 2024-09-18 at 08 50 24" src="https://github.com/user-attachments/assets/49796294-ea6c-4fca-90dc-adb473533320">
However, It doesnt mean you can't do it this way, but you will have to follow several steps manually, like here for Ceph https://medium.com/@keecheril.jobin/integrating-docker-with-ceph-56ce6c447d1b
I dont know exactly how ClusterFS works but it seems to require a custom docker volume driver, that is not yet supported by the deploy feature

**Reply by madalinignisca** [reply]
Fair enough.

**Reply by SquirrelDeveloper** [reply]
@madalinignisca However, I you have any example of an existing software implementing this service, I will take a look

**Reply by madalinignisca** [reply]
I can't provide any public examples, but you can think about clusters of databases, like MariaDB or Postgresql, with replication. Anything that requires data storage practically involves the equivalent of Stateful deployments from Kubernetes, where the pod and volume would be moved to another server in case the node goes down.

But your application makes sense, especially on single servers.
